lib/chef/provider/systemd_unit.rb

```diff
@@ -236,7 +236,13 @@ def enabled?
       def masked?
         # Note: masked-runtime is excluded, because runtime is volatile, and
         # because masked-runtime is not masked.
-        systemd_unit_status["UnitFileState"] == "masked"
+        # shell_out to get "masked" stdout.
+        shell_out!("systemctl list-unit-files | grep masked").stdout.each_line do |line|
```

Comment:
we're really trying to reduce the number of shell_out's that this provider needs to do.  is there a way to get this information out of the information that we already have?

Reply:
I have removed the extra `shell_out` from code and checked if a service is masked only on the basis of `UNIT_FILE_STATE` which can either be masked or bad in case a service is masked by the user. Let me know if this is an acceptable solution since we tested a few scenarios and looks like the `UNIT_FILE_STATE` check is good enough